app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -4,109 +4,151 @@ import android.content.Intent
+                if (result is CurrentSessionResult.Success) result.authSession.userId
+                else null
+            }
+            .distinctUntilChanged()
+            .shareIn(viewModelScope, SharingStarted.WhileSubscribed(), 1)
+
+        viewModelScope.launch {
+            launch { notificationManager.listenForMessageNotifications(userIdFlow) }
         }
     }
 
-    private val isUserLoggedIn = currentSession != null
-    var serverConfig: ServerConfig = ServerConfig.DEFAULT
-    private var ssoDeepLinkResult: DeepLinkResult.SSOLogin? = null
-
-    fun navigationArguments() =
-        if (ssoDeepLinkResult != null) {
-            listOf(serverConfig, ssoDeepLinkResult!!)
-        } else listOf(serverConfig)
+    fun navigationArguments() = navigationArguments.values.toList()
 
     fun startNavigationRoute() = when {
-        ssoDeepLinkResult is DeepLinkResult.SSOLogin -> NavigationItem.Login.getRouteWithArgs()
-        serverConfig.apiBaseUrl != ServerConfig.DEFAULT.apiBaseUrl -> NavigationItem.Login.getRouteWithArgs()
-        isUserLoggedIn -> NavigationItem.Home.getRouteWithArgs()
+        shouldGoToLogin() -> NavigationItem.Login.getRouteWithArgs()
+        shouldGoToHome() -> NavigationItem.Home.getRouteWithArgs()
         else -> NavigationItem.Welcome.getRouteWithArgs()
     }
 
-    private fun loadServerConfig(url: String) = runBlocking {
-        return@runBlocking when (val result = getServerConfigUseCase(url)) {
-            is GetServerConfigResult.Success -> result.serverConfig
-            else -> ServerConfig.DEFAULT
-        }
-    }
-
-    fun handleDeepLink(intent: Intent) {
-        intent.data?.let {
-            with(deepLinkProcessor(it)) {
+    fun handleDeepLink(intent: Intent?) {
+        intent?.data?.let {
+            val result = deepLinkProcessor(it)
+            with(result) {
                 when (this) {
                     is DeepLinkResult.CustomServerConfig ->
-                        serverConfig = loadServerConfig(url)
+                        navigationArguments.put(SERVER_CONFIG_ARG, loadServerConfig(url))
                     is DeepLinkResult.SSOLogin ->
-                        ssoDeepLinkResult = this
+                        navigationArguments.put(SSO_DEEPLINK_ARG, this)
+                    is DeepLinkResult.IncomingCall ->
+                        navigationArguments.put(INCOMING_CALL_CONVERSATION_ID_ARG, this.conversationsId)
                     DeepLinkResult.Unknown -> TODO()
                 }
             }
         }
     }
 
-    init {
-        viewModelScope.launch {
-            listenForNotificationsIfPossible()
+    /**
+     * Some of the deepLinks require to recreate Activity (Login, Welcome, etc.)
+     * Others needs just open some screen, without recreating (Conversation, IncomingCall, etc.)
+     *
+     * @return true if Activity need be to recreate, false - otherwise
+     */
+    fun handleDeepLinkOnNewIntent(intent: Intent?): Boolean {
```

Comment:
```suggestion
    /**
     * Some of the deepLinks require to recreate Activity (Login, Welcome, etc.)
     * Others need to just open some screen, without recreating it (Conversation, IncomingCall, etc.)
     *
     * @return true if Activity needs to be to recreated, false - otherwise
     */
    fun handleDeepLinkOnNewIntent(intent: Intent?): Boolean {
```

Reply:
Was this corrected?